cli/internal/cacheitem/tar.go

```diff
@@ -0,0 +1,91 @@
+// Adapted from https://github.com/moby/moby/blob/924edb948c2731df3b77697a8fcc85da3f6eef57/pkg/archive/archive.go
```

Comment:
Can we merge this under a different package and without waiting for the rest of the feature?

Reply:
Yeah, will split and ship today.

Reply:
#2116